netinterface.py

```diff
@@ -32,8 +32,8 @@ class NetInterface:
     Autoconf = 3
 
     def __init__(self, mode, hwaddr, ipaddr=None, netmask=None, gateway=None,
```

Comment:
all of the conditionals in this make this scream out for some heavy refactoring, splitting into different classes with static/dhcp helpers

Reply:
I agree, but I'd wager that this PR adds a new feature, to add refactoring on top of that would make it even more complex.
IMHO, a new PR (or several) should be created to refactor the code and perhaps adds some GH actions to enforce coding style and such.

Reply:
Typically the process is to refactor before adding the new feature and refactor after all changes to remove duplication and complexity.

Reply:
Just realising: a netintrface can be configured for both IPv4 and IPv6 so splitting in 2 classes is not possible.

Reply:
I've created a `NetinterfaceV6` class, inheriting from `Netinterface`, to split the `__init__` codes between the 2 classes.